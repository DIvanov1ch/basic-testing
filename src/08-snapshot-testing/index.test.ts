// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList([11, 22, 33, 44])).toStrictEqual({
      value: 11,
      next: {
        value: 22,
        next: {
          value: 33,
          next: { value: 44, next: { value: null, next: null } },
        },
      },
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    expect(generateLinkedList([11, 22, 33, 44])).toMatchSnapshot(`{
      value: 11,
      next: {
        value: 22,
        next: {
          value: 33,
          next: { value: 44, next: { value: null, next: null } },
        },
      },
    }`);
  });
});
