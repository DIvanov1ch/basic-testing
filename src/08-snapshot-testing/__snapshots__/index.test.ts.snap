// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2: {
      value: 11,
      next: {
        value: 22,
        next: {
          value: 33,
          next: { value: 44, next: { value: null, next: null } },
        },
      },
    } 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": null,
          "value": null,
        },
        "value": 44,
      },
      "value": 33,
    },
    "value": 22,
  },
  "value": 11,
}
`;
